gravitee-am-repository/gravitee-am-repository-jdbc/src/main/java/io/gravitee/am/repository/jdbc/management/api/JdbcUserRepository.java

```diff
@@ -568,13 +581,15 @@ public Flowable<User> findByIdIn(List<String> ids) {
         }
         return userRepository.findByIdIn(ids)
                 .map(this::toEntity)
-                .flatMap(user -> completeUser(user).toFlowable(), concurrentFlatmap);
+                .flatMap(user -> completeUser(user).toFlowable(), flatMapMaxConcurrency);
     }
 
     @Override
-    public Maybe<User> findById(ReferenceType referenceType, String referenceId, String userId) {
-        LOGGER.debug("findById({},{},{})", referenceType, referenceId, userId);
-        return userRepository.findById(referenceType.name(), referenceId, userId)
+    public Maybe<User> findById(Reference reference, UserId userId) {
+        LOGGER.debug("findById({},{})", reference, userId);
+        var criteria = userIdMatches(userId)
```

Comment:
not sure we should userIdMatch here, we should keep the "id" and don't manage the ext/src.